### PR TITLE
Deprecate exoscale_network "network_offering" attribute

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Exoscale API has changed and now no need to use NetworkOffering to create a Private-Network

### CODE REVIEW

**Comment by pierre-emmanuelJ** [comment]
Currently based on egoscale master branch

File: exoscale/resource_exoscale_network.go [hunk 1]
```diff
@@ -23,10 +23,6 @@ func resourceNetwork() *schema.Resource {
 			Required: true,
 			ForceNew: true,
 		},
-		"network_offering": {
```
Comment (falzm) [inline exoscale/resource_exoscale_network.go:26]: You should actually leave the attribute and add replace `Required: true` with `Removed: true`.

**Review by falzm — CHANGES_REQUESTED** [review]

**Comment by pierre-emmanuelJ** [comment]
```
$ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 GO111MODULE=on go test ./... -mod vendor -v  -timeout 120m
?   	github.com/terraform-providers/terraform-provider-exoscale	[no test files]
=== RUN   TestAccDatasourceComputeTemplate
--- PASS: TestAccDatasourceComputeTemplate (1.12s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestCheckResourceAttributes
--- PASS: TestCheckResourceAttributes (0.00s)
=== RUN   TestAccResourceAffinity
--- PASS: TestAccResourceAffinity (4.98s)
=== RUN   TestAccResourceCompute
--- PASS: TestAccResourceCompute (285.83s)
=== RUN   TestAccResourceDomainRecord
--- PASS: TestAccResourceDomainRecord (15.31s)
=== RUN   TestAccResourceDomain
--- PASS: TestAccResourceDomain (3.73s)
=== RUN   TestAccResourceIPAddress
--- PASS: TestAccResourceIPAddress (10.01s)
=== RUN   TestAccResourceNetwork
--- PASS: TestAccResourceNetwork (10.75s)
=== RUN   TestAccResourceNIC
--- PASS: TestAccResourceNIC (42.75s)
=== RUN   TestAccResourceSecondaryIPAddress
--- PASS: TestAccResourceSecondaryIPAddress (158.22s)
=== RUN   TestAccResourceSecurityGroupRule
--- PASS: TestAccResourceSecurityGroupRule (8.72s)
=== RUN   TestPreparePorts
--- PASS: TestPreparePorts (0.00s)
=== RUN   TestAccResourceSecurityGroupRules
--- PASS: TestAccResourceSecurityGroupRules (132.23s)
=== RUN   TestAccResourceSecurityGroup
--- PASS: TestAccResourceSecurityGroup (0.81s)
=== RUN   TestAccResourceSSHKeypair
--- PASS: TestAccResourceSSHKeypair (0.94s)
=== RUN   TestValidateStringKo
--- PASS: TestValidateStringKo (0.00s)
=== RUN   TestValidateStringOk
--- PASS: TestValidateStringOk (0.00s)
=== RUN   TestValidateRegexpKo
--- PASS: TestValidateRegexpKo (0.00s)
=== RUN   TestValidateRegexpOk
--- PASS: TestValidateRegexpOk (0.00s)
=== RUN   TestValidateUUIDKo
--- PASS: TestValidateUUIDKo (0.00s)
=== RUN   TestValidateUUIDOk
--- PASS: TestValidateUUIDOk (0.00s)
=== RUN   TestValidateIPv4StringNumber
--- PASS: TestValidateIPv4StringNumber (0.00s)
=== RUN   TestValidateIPv4StringNonIP
--- PASS: TestValidateIPv4StringNonIP (0.00s)
=== RUN   TestValidateIPv4StringOk
--- PASS: TestValidateIPv4StringOk (0.00s)
=== RUN   TestValidateIPv4StringKo
--- PASS: TestValidateIPv4StringKo (0.00s)
=== RUN   TestValidateIPv6StringNumber
--- PASS: TestValidateIPv6StringNumber (0.00s)
=== RUN   TestValidateIPv6StringNonIP
--- PASS: TestValidateIPv6StringNonIP (0.00s)
=== RUN   TestValidateIPv6StringKo
--- PASS: TestValidateIPv6StringKo (0.00s)
=== RUN   TestValidateIPv6StringOk
--- PASS: TestValidateIPv6StringOk (0.00s)
=== RUN   TestValidatePortRangeOk
--- PASS: TestValidatePortRangeOk (0.00s)
=== RUN   TestValidatePortRangeKo
--- PASS: TestValidatePortRangeKo (0.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-exoscale/exoscale	675.420s
?   	github.com/terraform-providers/terraform-provider-exoscale/version	[no test files]
```

File: exoscale/resource_exoscale_network.go (hunk 1 above)
Comment (pierre-emmanuelJ) [reply exoscale/resource_exoscale_network.go:26]: Better than removing attribute savagely 😆

**Comment by pierre-emmanuelJ** [comment]
The error if you set this attribute

```
config is invalid: "network_offering": [REMOVED] The new Exoscale API version doesn't need a Network Offering to create a Private Network.
```

**Review by falzm — APPROVED** [review]

File: exoscale/resource_exoscale_network.go (hunk 1 above)
Comment (falzm) [reply exoscale/resource_exoscale_network.go:26]: After discussion: even better is to deprecate the attribute.

**Review by illi-j — APPROVED** [review]
⛷️

**Comment by falzm** [comment]
Tested OK for with pre-existing configuration 👌

**Pull request merged**